src/Vault.sol

```diff
@@ -1095,44 +1292,29 @@ contract Vault is IERC4626, ERC20Permit, ILiquidationSource, IClaimable, Ownable
 
   /**
    * @notice Deposit assets and mint shares.
-   * @dev Will revert if the Vault is uncollateralized.
+   * @dev Will revert if the Vault is under-collateralized.
    *      So assets does not need to be converted to shares.
    * @param _assets The assets to deposit
    * @param _owner The owner of the assets
    * @param _receiver The receiver of the deposit shares
+   * @param _depositedAssets Assets deposited into the YieldVault
+   * @param _withdrawableAssets Assets withdrawable from the YieldVault
    * @return uint256 Amount of shares minted to `_receiver`
    */
   function _depositAssets(
     uint256 _assets,
     address _owner,
-    address _receiver
-  ) internal onlyVaultCollateralized returns (uint256) {
-    if (_assets > maxDeposit(_receiver)) {
-      revert DepositMoreThanMax(_receiver, _assets, maxDeposit(_receiver));
-    }
-
-    _deposit(_owner, _receiver, _assets);
-
-    return _assets;
-  }
-
-  /**
-   * @notice Deposit assets into the Vault and delegate to the sponsorship address.
-   * @dev There is no receiver parameter.
-   *      The calling address is the one depositing assets and receiving shares.
-   * @dev If the caller has not delegated to the sponsorship address yet, this function will.
-   * @param _assets Amount of assets to deposit
-   * @param _owner Address of the owner depositing `_assets`
-   * @return uint256 Amount of shares minted to `_receiver`.
-   */
-  function _sponsor(uint256 _assets, address _owner) internal returns (uint256) {
-    _depositAssets(_assets, _owner, _owner);
+    address _receiver,
+    uint256 _depositedAssets,
+    uint256 _withdrawableAssets
```

Comment:
After the latest changes, every time we are calling `_depositAssets`, we are also calling `totalAssets` and `totalSupply` exclusively for this function, so we can probably move the `totalAssets` and `totalSupply` calls into this function and remove the extra arguments.

Reply:
Fixed in this commit: https://github.com/GenerationSoftware/pt-v5-vault/commit/029a830aec2a6519a7c7375dec0ccde99d0e9ef5